runtime/tests/checker/interface_test.go

```diff
@@ -3988,4 +4030,62 @@ func TestCheckInheritedInterfacesSubtyping(t *testing.T) {
 
 		require.NoError(t, err)
 	})
+
+	t.Run("restricted anystruct reference subtyping", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {}
+
+            struct interface B: A  {}
+
+            struct interface C {}
+
+            struct S: B, C {}
+
+            // Case I: &{B, C} is a subtype of &{B}
+            fun foo(): &{B} {
+                var s: S{B, C} = S()
+                return &s as &{B, C}
+            }
+
+            // Case II: &{B} is a subtype of &{A}
+            fun bar(): &{A} {
+               var s: S{B} = S()
+               return &s as &{B}
+            }
+        `)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("restricted composite type reference subtyping", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t, `
+            struct interface A {}
+
+            struct interface B: A  {}
+
+            struct interface C {}
+
+            struct S: B, C {}
+
+            // Case I: &S{B, C} is a subtype of &S{B}
+            fun foo(): &S{B} {
+                var s: S{B, C} = S()
+                return &s as &S{B, C}
+            }
+
+            // Case II: &S{B} is a subtype of &S{A}
```

Comment:
Can we get a test for the combined case where `{B, C}` is a subtype of `{A}`?

Reply:
Added in fe0e730efa45ca8795fa6d5e7299ceb2d9dacb99